### PR TITLE
Customize chart colors

### PR DESCRIPTION
**Pull request by GziAzman**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [ ] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([french forum](https://community.gladysassistant.com/)/[english forum](https://en-community.gladysassistant.com/)) for testing before merging.
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

CF https://community.gladysassistant.com/t/feat-personnaliser-les-couleurs-des-graphiques/8774

### CODE REVIEW

File: front/src/components/boxs/chart/EditChart.jsx [hunk 1]
```diff
@@ -150,6 +197,11 @@ class EditChart extends Component {
   }
 
   render(props, { selectedDeviceFeaturesOptions, deviceOptions, loading, displayPreview }) {
+    const manyFeatures = selectedDeviceFeaturesOptions && selectedDeviceFeaturesOptions.length > 1;
+    const colorOptions = DEFAULT_COLORS.map((colorValue, i) => ({
+      value: colorValue,
+      label: DEFAULT_COLORS_NAME[i]
```
Comment (Pierre-Gilles) [inline front/src/components/boxs/chart/EditChart.jsx]: This should be translated !

File: server/models/dashboard.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ const boxesSchema = Joi.array().items(
       temperature_use_custom_value: Joi.boolean(),
       temperature_min: Joi.number(),
       temperature_max: Joi.number(),
+      colors: Joi.array().items(Joi.string().allow(null)),
```
Comment (Pierre-Gilles) [inline server/models/dashboard.js]: Why does a color could be null?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for this PR ! It works fine on my side, it's really great :) 

I have 2 small feedback but nothing really serious! Let me know if you need help to fix this

File: server/models/dashboard.js (hunk 2 above)
Comment (GziAzman) [reply server/models/dashboard.js]: In my first occurence (before your feedbacks), an user could let values empty, to fallback on default ones.
But now, default values are set in forms, so we don't need a nullable field anymore. 👌

**Comment by GziAzman** [comment]
@Pierre-Gilles Fixes done👌

**Review by Pierre-Gilles — APPROVED** [review]
Thanks for this great PR 🙌

Looks good to me!

**Pull request merged**